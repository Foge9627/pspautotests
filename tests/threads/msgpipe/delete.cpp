#include "shared.h"

inline void testDelete(const char *title, SceUID msgpipe) {
	int result = sceKernelDeleteMsgPipe(msgpipe);
	if (result == 0) {
		checkpoint("%s: OK", title);
	} else {
		checkpoint("%s: Failed (%08x)", title, result);
	}
}

extern "C" int main(int argc, char *argv[]) {
	SceUID msgpipe = sceKernelCreateMsgPipe("delete", PSP_MEMORY_PARTITION_USER, 0, 0x100, NULL);

	testDelete("Normal", msgpipe);
	testDelete("NULL", 0);
	testDelete("Invalid", 0xDEADBEEF);
	testDelete("Deleted", msgpipe);

	{
		msgpipe = sceKernelCreateMsgPipe("delete", PSP_MEMORY_PARTITION_USER, 0, 0x100, NULL);
		MsgPipeReceiveWaitThread wait_r("receiving thread", msgpipe);
		checkpoint("With receiving thread: %08x", sceKernelDeleteMsgPipe(msgpipe));
	}

	{
		msgpipe = sceKernelCreateMsgPipe("delete", PSP_MEMORY_PARTITION_USER, 0, 0x100, NULL);
		// Send something to fill it up.
		char msg[256];
		sceKernelSendMsgPipe(msgpipe, msg, sizeof(msg), 0, NULL, NULL);
		MsgPipeSendWaitThread wait_s("sending thread", msgpipe);
		checkpoint("With sending thread: %08x", sceKernelDeleteMsgPipe(msgpipe));
	}

	return 0;
}