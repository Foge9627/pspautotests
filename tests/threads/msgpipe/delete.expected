[x] Normal: OK
[x] NULL: Failed (8002019e)
[x] Invalid: Failed (8002019e)
[x] Deleted: Failed (8002019e)
[r]   ** started receiving thread
[x] With receiving thread: 00000000
[r]   ** receiving thread got result: 800201b5, received = 00000000
[r]   ** stopped receiving thread
[r]   ** started sending thread
[x] With sending thread: 00000000
[r]   ** sending thread got result: 800201b5, sent = 00000000
[r]   ** stopped sending thread
